packages/suite-data/src/guide/parser.ts

```diff
@@ -0,0 +1,214 @@
+    // which doesn't contain a README.md.
+    // Strip last 3 chars because locale (always two letters) and separator.
+    if (path.slice(0, -3) === TMP) {
+        // A sensible fallback for title of the root Node.
+        // Won't be shown in the UI anyway.
+        return 'Suite Guide';
+    }
+    // For pages with children (i.e. Categories)
+    // GitBook creates a directory with README.md.
+    // Parse the title from there.
+    return parsePageTitle(join(path, 'README.md'));
+};
+
+/** Returns a title of given page. */
+const parsePageTitle = (path: string): string => {
+    const doc = fs.readFileSync(path);
+    // In GitBook a title is edited in a dedicated field
+    // and its content is added to the markdown file as level-1 heading.
+    // Find the line that starts with a single hash.
+    const title = doc
+        .toString()
+        .match(/^# (.+$)/m)?.[1]
+        .trim();
+    if (title === undefined) {
+        throw new Error(`Could not parse title from ${path}.`);
+    }
+    return title;
+};
+
+/** Converts given path to locale- and environment-agnostic id. */
+const pathToId = (path: string): string => {
+    // Remove the environment-specific part of the path
+    // and the locale.
+    const id = path.slice(TMP.length + 3);
+    if (id.length !== 0) {
+        return id;
+    }
+    // Use `/` as id for the root Node.
+    return '/';
+};
+
+/**
+ * Given a path to a file-tree of localized GitBook content
+ * recursively traverses and parses it and returns its
+ * object representation.
+ *
+ * The returned representation is localized. Its
+ * language must be set by the `locale` param as it
+ * cannot be inferred from the content itself.
+ *
+ * @param path to root of the file-tree to start at.
+ * @param locale locale of the given tree.
+ * @returns object representation of the given tree.
+ */
+const parseTree = (path: string, locale: string): Node => {
+    if (isDirectory(path)) {
+        const children = fs
+            .readdirSync(path)
+            // Ignore READMEs as they're only generated by GB for pages that have children
+            // while we're only interested in content of leaf pages i.e. pages without children.
+            .filter(child => child !== 'README.md')
+            .map(child => parseTree(join(path, child), locale));
+
+        return {
+            type: 'category',
+            id: pathToId(path),
+            locales: [locale],
+            title: {
+                [locale]: parseDirTitle(path),
+            },
+            children: children,
+        };
+    }
+
+    return {
+        type: 'page',
+        id: pathToId(path),
+        locales: [locale],
+        title: {
+            [locale]: parsePageTitle(path),
+        },
+    };
+};
+
+/** Returns true if left and right are variants of the same content. */
+const match = (left: Node, right: Node): boolean => {
+    return left.type === right.type && left.id === right.id;
+};
```

Comment:
Suggestions does not work today for me, sorry.

`const match = (left: Node, right: Node): boolean => left.type === right.type && left.id === right.id;`

Eslint error

Reply:
I would prefer to change the eslint rule instead... it's easier to extend the function and easier to read the changelog when the function is extended...
But of course it doesn't really matter, only the consistency is important.

Reply:
My IDE doesn't show me these linting errors. I don't say they're wrong (quite the opposite!) but I'd like to setup the linter properly before spending time on linting. By properly I mean with explicit config that will be picked up by everyone's IDE and that will be enforced in CI as well. I created an issue for that https://github.com/trezor/trezor-suite/issues/3870. There's IMO no reason to spend time on shedding on styling now (or ever actually). I'd propose to resolve this and tackle it once we'll setup the linter. What do you think, guys?

Reply:
I didn't know the linter wasn't set up in suite-data... +1 for #3870